tests/gaco.cpp

```diff
@@ -240,10 +241,34 @@ struct udp_inf {
     unsigned int m_dim;
 };
 
+struct udp_nan {
+    
+    /// Fitness
+    vector_double fitness(const vector_double &) const
+    {
+        double nan = std::numeric_limits<double>::quiet_NaN();
+        return {nan};
+    }
+    vector_double::size_type get_nobj() const
+    {
+        return 1u;
+    }
+    
+    /// Problem bounds
+    std::pair<vector_double, vector_double> get_bounds() const
+    {
+        return {{0., 0.}, {1., 1.}};
+    }
+    /// Problem dimensions
+    unsigned int m_dim;
+};
+
 BOOST_AUTO_TEST_CASE(test_for_inf)
 {
     problem prob(udp_inf);
+    problem prob_2(udp_nan);
```

Comment:
I just realized that these two lines are completely useless. I will erase them in the next commit